static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [isAtStart, setIsAtStart] = useState<boolean>(true);
+  const [isAtEnd, setIsAtEnd] = useState<boolean>(false);
+  const [showArrows, setShowArrows] = useState<boolean>(true);
+
+  const setStartAndEnd = (scrollLeft: number, totalScrollDist: number) => {
+    // scrollLeft: the amount scrolled horizontally.
+    // start at 0 and end at totalScrollDist
+
+    // totalScrollDist = scrollWidth - clientWidth
+    // scrollWidth: total element width including overflow
+    // clientWidth: width of the element excluding overflow
+    if (scrollLeft <= 0) {
+      setIsAtStart(true);
+      setIsAtEnd(false);
+    } else if (scrollLeft >= totalScrollDist) {
+      setIsAtStart(false);
+      setIsAtEnd(true);
+    } else {
+      setIsAtStart(false);
+      setIsAtEnd(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const contentWidth = ref.current.scrollWidth - ref.current.clientWidth;
+    if (contentWidth === 0) {
+      setShowArrows(false);
+    }
+
+    const scrollLeft = ref.current.scrollLeft;
+    setStartAndEnd(scrollLeft, contentWidth);
+  }, [ref]);
+
+  const handleScroll = (direction: string) => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const itemWidth = parseInt(getComputedStyle(ref.current.children[0]).width, 10);
+
+      let scrollDist = 0;
+      if (direction === 'left') {
+        scrollDist = scrollLeft - itemWidth;
+      } else if (direction === 'right') {
+        scrollDist = scrollLeft + itemWidth;
+      }
+      ref.current.scrollLeft = scrollDist;
+    });
+  };
+
+  const setArrows = () => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const totalScrollDist = ref.current.scrollWidth - ref.current.clientWidth;
+
+      setStartAndEnd(scrollLeft, totalScrollDist);
+    });
+  };
+
+  return (
+    <CarouselContainer>
+      <CarouselItems ref={ref} onScroll={setArrows}>
+        {children}
+      </CarouselItems>
+      {showArrows && !isAtStart && (
+        <CircledArrow onClick={() => handleScroll('left')} direction="left" />
+      )}
+      {showArrows && !isAtEnd && (
+        <CircledArrow onClick={() => handleScroll('right')} direction="right" />
+      )}
+    </CarouselContainer>
+  );
+}
+
+const CarouselContainer = styled('div')`
+  position: relative;
+`;
+
+const CarouselItems = styled('div')`
+  display: flex;
+  overflow-x: scroll;
+  scroll-behavior: smooth;
+  &::-webkit-scrollbar {
+    background-color: transparent;
+    height: 8px;
+  }
+  &::-webkit-scrollbar-thumb {
+    background: ${p => p.theme.gray400};
+    border-radius: 8px;
+  }
+`;
+
+type ArrowProps = {
```

Comment:
```suggestion
type CircledArrowProps = {
```

Though I would probably rename the component to `ScrollButton` and name this `ScrollButtonProps`